dipy/segment/tests/test_quickbundles.py

```diff
@@ -194,6 +194,19 @@ def test_quickbundles_with_not_order_invariant_metric():
     assert_array_equal(clusters[0].centroid, streamline)
 
 
+def test_quickbundles_with_not_order_invariant_metric_multiple_clusters():
+    metric = dipymetric.AveragePointwiseEuclideanMetric()
+    qb = QuickBundles(threshold=0.1, metric=metric)
+
+    streamline = np.arange(10*3, dtype=dtype).reshape((-1, 3))
+    streamlines = [streamline, streamline[::-1]]
+
+    clusters = qb.cluster(streamlines)
+    assert_equal(len(clusters), 2)
```

Comment:
It is not clear to me why you expect two clusters here. Both streamlines are the same, they just start from different ends. If you truly want to differentiate these two streamlines, you don't want to use QuickBundles because QB relies on the flip to merge streamlines correctly (i.e. to form the centroids).

Let me know if I didn't understand what you are trying to accomplish.